target/linux/brcm63xx/dts/agpwi.dts

```diff
@@ -0,0 +1,125 @@
+/dts-v1/;
+
+#include "bcm6328.dtsi"
+
+#include <dt-bindings/input/input.h>
+
+/ {
+	model = "ADB P.DG A4001N A-000-1A1-AX";
+	compatible = "adb,agpwi", "brcm,bcm6328";
```

Comment:
You should decide on which primary name you prefer for that device. Either use the Telecom Italia name, or the ADB name. But please don't use a different name here compared to the image/Makefile definitions. What's it with the "agpwi" BTW? I'd actually prefer if we could build a compatible based on the telecom italia model name.

If that's decide, note that you will have to consistently use that for compatible, base-files and DTS name then.

The DTS file should be named soc-vendor-model.dts, so with your current compatible it would be bcm6328-adb-agpwi.dts

Reply:
This router, as you understood, has three names:

the ADB one univocally identifies it and the PCB inside.

the AGPWI one univocally identifies it and the PCB inside too and it's derived from it's stock firmware name. Around the web people commonly refer to this router as AGPWI. The same exact thing was done for the Pirelli's AGPF router. OpenWrt's official images are also named as AGPF but the real Pirelli and Telecom italia's name is not AGPF, it is the Telecom Italia's firmware name for that router.

the Telecom Italia's name (now TIM) for the AGPWI does not univocally identify this router because another router commonly named as AGTWI has got the same Telecom Italia's name. These two routers are estetically different and made from two different vendors but, wird enought, the two board seems to be compatible: In the AGTWI (Technicolor TG582n) that mounts the 8MB flash (DANT-T PCB revision) you can flash the official OpenWrt image for the ADB P.DG A4001N A-000-1A1-AE (openwrt-x.x.x-brcm63xx-generic-A4001N-squashfs-cfe.bin) like for my AGPWI.
I have got the AGTWI board too but I did not flash openwrt yet and so I don't know if the image I modified for my AGPWI would work too for my AGTWI DANT-T revision or if the OpenWrt's openwrt-x.x.x-brcm63xx-generic-A4001N-squashfs-cfe.bin will work fine.
The OpenWrt's openwrt-x.x.x-brcm63xx-generic-A4001N-squashfs-cfe.bin will work for sure in both of these routers but in the AGPWI the LEDs will be wrongly mapped in the device tree and I don't know yet if my fix for the AGPWI will work for the AGTWI DANT-T too or if the actual OpenWrt image works just fine for its LEDs.

Now you can see that I'm really confused too in how to call this router in the files:
I would like to let the OpenWrt image name be AGPWI (and its internal files too for shortness)  so that people can search on the web for the AGPWI image, but I want LUCI to display the ADB name that for me is the real correct one.

So, what should I do?

Reply:
Let me think about it. Just note that after my recent update the vendor name will be included in the image name, so you would have e.g. openwrt-x.x.x-brcm63xx-generic-adb_agpwi-squashfs-cfe.bin

Reply:
In that case the vendor name should be Telecom Italia because AGPWI is The Telecom Italia's firmware name that has notthing to do with the ADB's firmware.

I think that "openwrt-x.x.x-brcm63xx-generic-Telecom_Italia_agpwi-squashfs-cfe.bin" should be a proper image name: at least people can distinguish between Telecom Italia's AGTWI and AGPWI. Then, in the future, when I'll test this fix to the AGTWI too, I'll maybe discover that this fix works for the AGTWI as well and the image name can be changed in something like "openwrt-x.x.x-brcm63xx-generic-Telecom_Italia_ADSL2+_Wi-Fi_N-squashfs-cfe.bin" but at that point then people can't search for the AGPWI or AGPWI image anymore...
Furthermore "ADSL2+ Wi-Fi N" looks to me as a too generic name for a router...

Anyway I think I will wait for your decision about namings of this router before I apply the modifications you requested to me

Reply:
I think I found a solution:

I noticed that the OpenWrt image name for the AGPF is called like so because It seems that Pirelli did not give any name to that router leaving the Alice (Telecom Italia) name that is the same for both the AGPF and the AGIA router. People, to distinguish between these routers, started to use these two names AGPF and AGIA.

This is not the case for the AGPWI for wich on the bottom of the router you can find both the ADB and Telecom Italia's names.

I think I will simpli call all the files with the full ADB name like so for the dts:
"bcm6328-adb-a4001n-a-000-1a1-ax.dts"
so the OpenWrt image name will be:
"openwrt-x.x.x-brcm63xx-generic-adb-a4001n-a-000-1a1-ax-squashfs-cfe.bin"
I will not put Telecom Italia ADSL2+ Wi-Fi N AGPWI name nowhere around the sources so that people who own this router with no Telecom Italia brand name will be facilitated.

Please, let me know what do you thing of this solution :)

Reply:
> I think I will simply call all the files with the full ADB name like so for the dts:
> "bcm6328-adb-a4001n-a-000-1a1-ax.dts"
> so the OpenWrt image name will be:
> "openwrt-x.x.x-brcm63xx-generic-adb-a4001n-a-000-1a1-ax-squashfs-cfe.bin"

That is the correct solution for sure. It would also be consistent with other newer devices from TIM which also have an AGxxx firmware name, an original marketing name, a vendor model name, and a vendor board name. For instance, "TIM Smart Modem Plus", firmware AGTEF, vendor model "DGA4130", board VBNT-K would become openwrt-x.x.x-brcm63xx-generic-technicolor-dga4130-vbnt-k-squashfs-cfe.bin, and that would also work on the same device from other service providers where no "Smart Modem Plus" or "AGTEF" names appear anywhere.